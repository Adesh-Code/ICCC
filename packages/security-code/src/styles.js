const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
