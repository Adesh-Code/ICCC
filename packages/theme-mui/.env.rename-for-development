# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE