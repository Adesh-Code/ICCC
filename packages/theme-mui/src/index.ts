// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
