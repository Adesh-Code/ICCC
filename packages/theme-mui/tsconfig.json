{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
