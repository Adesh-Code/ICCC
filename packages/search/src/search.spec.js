import React from "react";
import {
  fireEvent,
  render,
  screen,
  waitFor,
  within
} from "@testing-library/react";
import Search from "./search";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const onInputChangeMock = jest.fn();

const arrange = (props) => {
  const args = {
    id: "test-search",
    options: ["Alfa", "Bravo", "Charlie", "Delta", "Echo"],
    ...props
  };

  return render(
    <ThemeProvider theme={light}>
      <Search
        {...args}
        onInputChange={(event, value, reason) => onInputChangeMock(value)}
      />
    </ThemeProvider>
  );
};

describe("Search", () => {
  it("mounts the search component", () => {
    arrange();
    const search = screen.getByRole("combobox");
    expect(search).toMatchSnapshot();
  });

  it("doesn't automatically display dropdown on focus ", () => {
    arrange();
    const search = screen.getByRole("combobox");
    search.focus();
    expect(screen.queryByText("Alfa")).not.toBeInTheDocument();
  });

  it("displays dropdown when results of search query are found", async () => {
    const { container } = arrange();
    const search = container.firstChild;
    const input = within(search).getByRole("combobox");
    fireEvent.change(input, { target: { value: "a" } });
    expect(onInputChangeMock).toHaveBeenCalledWith("a");
    await waitFor(() => screen.getByRole("listbox"));
    expect(screen.getByText("Alfa")).toBeInTheDocument();
    // select a valid option from search dropdown
    fireEvent.click(screen.getByText("Alfa"));
    expect(onInputChangeMock).toHaveBeenCalledWith("Alfa");
  });

  it("accepts classnames", () => {
    const { container } = arrange({
      className: "foo"
    });
    const search = container.firstChild;
    expect(search).toHaveClass("foo");
  });
});
