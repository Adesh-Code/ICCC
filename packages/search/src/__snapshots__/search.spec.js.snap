// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search mounts the search component 1`] = `
<input
  aria-autocomplete="list"
  aria-expanded="false"
  aria-invalid="false"
  autocapitalize="none"
  autocomplete="off"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart DhigSearch--input emotion-0"
  id="test-search"
  placeholder="Search..."
  role="combobox"
  spellcheck="false"
  type="text"
  value=""
/>
`;
